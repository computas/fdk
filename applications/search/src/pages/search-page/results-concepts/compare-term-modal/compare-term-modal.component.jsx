import React from 'react';
import { Modal, ModalBody, ModalHeader } from 'reactstrap';
import PropTypes from 'prop-types';

import localization from '../../../../lib/localization';
import { CompareTermModalContent } from './compare-term-modal-content/compare-term-modal-content.component';
import './compare-term-modal.scss';

const ReactGA = require('react-ga');

export class CompareTermModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showModal: false,
      showBar: true
    };
    this.open = this.open.bind(this);
    this.close = this.close.bind(this);
  }

  close() {
    this.setState({
      showModal: false,
      showBar: true
    });
  }

  open(type) {
    if (type === 'BOTTOM') {
      ReactGA.event({
        category: 'Begrep',
        action: 'Sammenligningsvindu',
        label: 'Sticky button'
      });
    } else {
      ReactGA.event({
        category: 'Begrep',
        action: 'Sammenligningsvindu',
        label: 'Left menu button'
      });
    }
    ReactGA.modalview('/concepts/compare');
    this.setState({
      showModal: true,
      showBar: false
    });
  }

  render() {
    let cols = 'col-md-';
    switch (12 / this.props.terms.length) {
      case 6:
        cols += '6';
        break;
      case 4:
        cols += '4';
        break;
      default:
        cols += '6';
        break;
    }

    const removeTerms = items =>
      items.map((item, index) => (
        <div className={cols} key={`remove-${index}${item.uri}`}>
          <button
            className="fdk-concepts-compare-rm-term fdk-button-small fdk-fullwidth mt-5 fdk-modal-button"
            onClick={() => {
              this.props.handleDeleteTerm(index);
            }}
          >
            <i className="fa fa-times fdk-color0" />
            &nbsp;
            {localization.compare.remove}
          </button>
        </div>
      ));

    const bottomModalLabel = (
      <div>
        <button
          className="bottom-modal-label fdk-button fdk-button-cta fdk-modal-button"
          onClick={() => {
            this.open('BOTTOM');
          }}
        >
          <i className="fa fa-chevron-up fdk-color0" />
          &nbsp;
          {localization.compare.added +
            this.props.terms.length +
            localization.compare.toCompare}
        </button>
      </div>
    );

    return (
      <div>
        <button
          className="fdk-button fdk-button-cta fdk-fullwidth"
          onClick={this.open}
        >
          <i className="fa fa-chevron-up fdk-color0" />
          &nbsp;
          {localization.compare.openCompare}
        </button>
        <Modal
          isOpen={this.state.showModal}
          toggle={this.close}
          className="fdk-modal modal-lg"
        >
          <ModalHeader toggle={this.close}>
            <i className="fa fa-chevron-down fdk-fa-left fa-lg fdk-color0" />
            &nbsp;
            {localization.compare.compare}
          </ModalHeader>
          <ModalBody>
            <CompareTermModalContent terms={this.props.terms} cols={cols} />
            <div className="row">{removeTerms(this.props.terms)}</div>
          </ModalBody>
        </Modal>
        {this.state.showBar && this.props.terms.length > 0
          ? bottomModalLabel
          : null}
      </div>
    );
  }
}

CompareTermModal.defaultProps = {
  terms: null
};

CompareTermModal.propTypes = {
  terms: PropTypes.array,
  handleDeleteTerm: PropTypes.func.isRequired
};
