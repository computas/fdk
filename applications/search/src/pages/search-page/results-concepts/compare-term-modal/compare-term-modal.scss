@import '../../../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.fdk-modal {
  width: 90% !important;
  top: 10%;
}
.modal-header {  
  background-color: $color1; 
  color: $color0;
  padding-left: 2em !important;
  padding-right: 2em !important;
  h4 {    
    color: inherit;
  }
  h4:hover {
    cursor: pointer
  }
}

.modal-body {
  background-color: $color0; 
  color: $color1;
  padding: 1.5em !important;
  h5 {
    color: inherit;
  }
}

.fdk-modal-button {
  font-weight: 600;
}

.bottom-modal-label {
  position: fixed !important;
  width: 76%;
  height: 3.5em;
  left: 12%;
  bottom: 0;
  z-index: 100;
  margin-bottom: 0 !important;
  border-bottom: 0 !important;
}