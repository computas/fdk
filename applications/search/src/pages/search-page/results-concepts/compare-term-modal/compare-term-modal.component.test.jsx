import React from 'react';
import { shallow } from 'enzyme';
import { CompareTermModal } from './compare-term-modal.component';
import concepts from '../../../../../test/fixtures/concepts';

let handleDeleteTerm;
let defaultProps;
let wrapper;

beforeEach(() => {
  handleDeleteTerm = jest.fn();
  defaultProps = {
    terms: [concepts.hits.hits[0]._source, concepts.hits.hits[1]._source],
    handleDeleteTerm
  };
  wrapper = shallow(<CompareTermModal {...defaultProps} />);
});

test('should render CompareTermModal correctly with props', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render CompareTermModal correctly with one term', () => {
  const deleteTerm = jest.fn();
  const props = {
    terms: [concepts.hits.hits[0]._source],
    handleDeleteTerm: deleteTerm
  };
  const wrapperWithProp = shallow(<CompareTermModal {...props} />);
  expect(wrapperWithProp).toMatchSnapshot();
});

test('should handle open correctly', () => {
  wrapper.find('button.bottom-modal-label').simulate('click');
  expect(wrapper.state('showModal')).toBe(true);
  expect(wrapper.state('showBar')).toBe(false);
});

test('should handle modal onClick', () => {
  wrapper.find('ModalHeader').prop('onClick');
  expect(wrapper.state('showModal')).toBe(false);
  expect(wrapper.state('showBar')).toBe(true);
});

test('should handle onChange', () => {
  wrapper
    .find('button.fdk-concepts-compare-rm-term')
    .at(0)
    .prop('onClick')();
  expect(handleDeleteTerm).toHaveBeenLastCalledWith(0);
});
