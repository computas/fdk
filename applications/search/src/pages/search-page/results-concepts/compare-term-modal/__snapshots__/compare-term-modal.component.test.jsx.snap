// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CompareTermModal correctly with one term 1`] = `
<div>
  <button
    className="fdk-button fdk-button-cta fdk-fullwidth"
    onClick={[Function]}
  >
    <i
      className="fa fa-chevron-up fdk-color0"
    />
     
    Show comparison
  </button>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    className="fdk-modal modal-lg"
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      closeAriaLabel="Close"
      tag="h5"
      toggle={[Function]}
      wrapTag="div"
    >
      <i
        className="fa fa-chevron-down fdk-fa-left fa-lg fdk-color0"
      />
       
      Comparison
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      <CompareTermModalContent
        cols="col-md-6"
        terms={
          Array [
            Object {
              "creator": Object {
                "id": "974760673",
                "naeringskode": Object {
                  "code": "84.110",
                  "prefLabel": Object {
                    "no": "Generell offentlig administrasjon",
                  },
                  "uri": "http://www.ssb.no/nace/sn2007/84.110",
                },
                "name": "Brønnøysundregistrene",
                "orgPath": "/STAT/912660680/974760673",
                "organisasjonsform": "ORGL",
                "overordnetEnhet": "912660680",
                "sektorkode": Object {
                  "code": "6100",
                  "prefLabel": Object {
                    "no": "Statsforvaltningen",
                  },
                  "uri": "http://www.brreg.no/sektorkode/6100",
                },
                "type": "no.dcat.datastore.domain.dcat.Publisher",
                "uri": "http://data.brreg.no/enhetsregisteret/enhet/974760673",
                "valid": true,
              },
              "datasets": Array [
                Object {
                  "description": Object {
                    "en": "Central Coordinating Register for Legal Entities is a register containing information on all legal entities in Norway - commercial enterprises and government agencies. Includes also businesses sole proprietorships, associations and other economic entities without registration duty, who have chosen to join the CCR on a voluntary basis.",
                    "nb": "Enhetsregisteret inneholder oversikt over alle registrerte enheter Norge innen næringsliv, offentlig og frivillig sektor. Enhetsregisteret tildeler organisasjonsnummer, og samordner opplysninger om næringsliv og offentlige etater som finnes i ulike offentlige registre.",
                  },
                  "id": "b3abe917-f2c6-4bda-8e17-8982f95ba11f",
                  "title": Object {
                    "en": "Central Coordinating Register for Legal Entities",
                    "nb": "Enhetsregisteret",
                  },
                  "uri": "http://data.brreg.no/datakatalog/dataset/974760673/2",
                },
              ],
              "definition": Object {
                "no": "alt som er registrert med et organisasjonsnummer ",
              },
              "identifier": "https://data-david.github.io/Begrep/begrep/Enhet",
              "inScheme": Array [
                "http://data-david.github.io/vokabular/Befolkning",
              ],
              "note": Object {
                "no": "Alle hovedenheter, underenheter og organisasjonsledd som er identifisert med et organisasjonsnummer.",
              },
              "prefLabel": Object {
                "no": "enhet",
              },
              "source": "https://jira.brreg.no/browse/BEGREP-208",
              "uri": "https://data-david.github.io/Begrep/begrep/Enhet",
            },
          ]
        }
      />
      <div
        className="row"
      >
        <div
          className="col-md-6"
          key="remove-0https://data-david.github.io/Begrep/begrep/Enhet"
        >
          <button
            className="fdk-concepts-compare-rm-term fdk-button-small fdk-fullwidth mt-5 fdk-modal-button"
            onClick={[Function]}
          >
            <i
              className="fa fa-times fdk-color0"
            />
             
            Remove concept
          </button>
        </div>
      </div>
    </ModalBody>
  </Modal>
  <div>
    <button
      className="bottom-modal-label fdk-button fdk-button-cta fdk-modal-button"
      onClick={[Function]}
    >
      <i
        className="fa fa-chevron-up fdk-color0"
      />
       
      You have added 1 concepts to comparison - click to show
    </button>
  </div>
</div>
`;

exports[`should render CompareTermModal correctly with props 1`] = `
<div>
  <button
    className="fdk-button fdk-button-cta fdk-fullwidth"
    onClick={[Function]}
  >
    <i
      className="fa fa-chevron-up fdk-color0"
    />
     
    Show comparison
  </button>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    className="fdk-modal modal-lg"
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      closeAriaLabel="Close"
      tag="h5"
      toggle={[Function]}
      wrapTag="div"
    >
      <i
        className="fa fa-chevron-down fdk-fa-left fa-lg fdk-color0"
      />
       
      Comparison
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      <CompareTermModalContent
        cols="col-md-6"
        terms={
          Array [
            Object {
              "creator": Object {
                "id": "974760673",
                "naeringskode": Object {
                  "code": "84.110",
                  "prefLabel": Object {
                    "no": "Generell offentlig administrasjon",
                  },
                  "uri": "http://www.ssb.no/nace/sn2007/84.110",
                },
                "name": "Brønnøysundregistrene",
                "orgPath": "/STAT/912660680/974760673",
                "organisasjonsform": "ORGL",
                "overordnetEnhet": "912660680",
                "sektorkode": Object {
                  "code": "6100",
                  "prefLabel": Object {
                    "no": "Statsforvaltningen",
                  },
                  "uri": "http://www.brreg.no/sektorkode/6100",
                },
                "type": "no.dcat.datastore.domain.dcat.Publisher",
                "uri": "http://data.brreg.no/enhetsregisteret/enhet/974760673",
                "valid": true,
              },
              "datasets": Array [
                Object {
                  "description": Object {
                    "en": "Central Coordinating Register for Legal Entities is a register containing information on all legal entities in Norway - commercial enterprises and government agencies. Includes also businesses sole proprietorships, associations and other economic entities without registration duty, who have chosen to join the CCR on a voluntary basis.",
                    "nb": "Enhetsregisteret inneholder oversikt over alle registrerte enheter Norge innen næringsliv, offentlig og frivillig sektor. Enhetsregisteret tildeler organisasjonsnummer, og samordner opplysninger om næringsliv og offentlige etater som finnes i ulike offentlige registre.",
                  },
                  "id": "b3abe917-f2c6-4bda-8e17-8982f95ba11f",
                  "title": Object {
                    "en": "Central Coordinating Register for Legal Entities",
                    "nb": "Enhetsregisteret",
                  },
                  "uri": "http://data.brreg.no/datakatalog/dataset/974760673/2",
                },
              ],
              "definition": Object {
                "no": "alt som er registrert med et organisasjonsnummer ",
              },
              "identifier": "https://data-david.github.io/Begrep/begrep/Enhet",
              "inScheme": Array [
                "http://data-david.github.io/vokabular/Befolkning",
              ],
              "note": Object {
                "no": "Alle hovedenheter, underenheter og organisasjonsledd som er identifisert med et organisasjonsnummer.",
              },
              "prefLabel": Object {
                "no": "enhet",
              },
              "source": "https://jira.brreg.no/browse/BEGREP-208",
              "uri": "https://data-david.github.io/Begrep/begrep/Enhet",
            },
            Object {
              "altLabel": Array [
                Object {
                  "no": "Orgid",
                },
                Object {
                  "no": "Orgnr",
                },
              ],
              "datasets": Array [
                Object {
                  "description": Object {
                    "nb": "Akvakulturregisteret inneholder oversikt over akvakulturtillatelser og enkelte vesentlige vedtak knyttet til disse. I tillegg inneholder registeret en oversikt over overføringer, pantsettelser og andre rettigheter som er tinglyst på tillatelsene.",
                  },
                  "id": "631b8f86-530e-44f0-b0b6-cad948c6b9b6",
                  "title": Object {
                    "en": "Aquaculture register",
                    "nb": "Akvakulturregisteret",
                  },
                  "uri": "http://data.brreg.no/datakatalog/dataset/974760673/5",
                },
                Object {
                  "description": Object {
                    "en": "Register of clubs and associations. The purpose is to improve and simplify the interaction between NGOs and public authorities. The Registry will ensure systematic information that can strengthen the legitimacy and knowledge of the voluntary activity. Managed by Brønnøysund Registre",
                    "nb": "Frivillighetsregisteret har oversikt over lag og foreninger i Norge, og har som formål å forenkle samhandlingen mellom frivillige organisasjoner og offentlig sektor. De som registrerer seg i Frivillighetsregisteret kan delta i grasrotandelen til Norsk Tipping, og motta/få momskompensasjon.",
                  },
                  "id": "6d71993e-4c9b-4d5b-9f57-c7f2cea1037b",
                  "title": Object {
                    "nb": "Frivillighetsregisteret",
                  },
                  "uri": "http://data.brreg.no/datakatalog/dataset/974760673/10",
                },
                Object {
                  "description": Object {
                    "nb": "Partiregisteret er et register over politiske parti i Norge. Hovedformålet med registeret er å gi part anledning til å skaffe seg enerett på partinavn.",
                  },
                  "id": "4abdf712-1c9d-49b9-b5f9-6c67a8983eda",
                  "title": Object {
                    "nb": "Partiregisteret",
                  },
                  "uri": "http://data.brreg.no/datakatalog/dataset/974760673/7",
                },
                Object {
                  "description": Object {
                    "nb": "Registeret for utøvere av alternativ behandling inneholder opplysninger om registrerte utøvere og godkjente utøverorganisasjoner. Registeret skal bidra med å vareta pasientvernet og forbrukerrettighetene for den som oppsøker en registrert alternativ behandler. I tillegg skal registeret medvirke til seriøsitet og ordnede forretningsvilkår",
                  },
                  "id": "e063a919-d4ba-4b78-9fe3-0c114cbcef68",
                  "title": Object {
                    "nb": "Registeret for utøvere av alternativ behandling",
                  },
                  "uri": "http://data.brreg.no/datakatalog/dataset/974760673/16",
                },
                Object {
                  "description": Object {
                    "en": "The Register of Business Enterprises registers Norwegian and foreign businesses in Norway, ensuring legal protection and financial overview. All businesses – with confining as well as complete responsibility – are obliged to register in the Register of Business Enterprises. The same applies to sole proprietorships engaging in trade with purchased goods or having more than five fully employed personell. Other sole proprietorships may register on a voluntary basis.",
                    "nb": "Foretaksregisteret registrerer alle norske og utenlandske foretak i Norge, og skal sikre rettsvern og økonomisk oversikt. Enkeltpersonforetak som ikke driver handel med innkjøpte varer eller har mindre enn 5 ansatte i full stilling har ikke registreringsplikt, men har registreringsrett. Foretaksregisteret skal sikre rettsvern og økonomisk oversikt, og er en viktig kilde til informasjon om norsk næringsliv.",
                  },
                  "id": "97093ad6-661a-46f8-b4f6-681998661d03",
                  "title": Object {
                    "en": "Register of Business Enterprises",
                    "nb": "Foretaksregisteret",
                  },
                  "uri": "http://data.brreg.no/datakatalog/dataset/974760673/1",
                },
                Object {
                  "description": Object {
                    "nb": "Løsøreregisteret er et tinglysingsregister for rettigheter og pant i løsøre. En tinglyst rettighet eller pant, gir rettsvern overfor kreditorer. Løsøreregisteret omfatter ikke tinglysning i fast eiendom, skip og luftfartøyer.",
                  },
                  "id": "510ebd85-15f5-40b8-8bad-dfb92234ee2a",
                  "title": Object {
                    "en": "Register of Mortgaged Movable Property",
                    "nb": "Løsøreregisteret",
                  },
                  "uri": "http://data.brreg.no/datakatalog/dataset/974760673/3",
                },
                Object {
                  "description": Object {
                    "nb": "Registeret for offentlig støtte (ROFS) er et nasjonalt register for offentlige støttetildelinger som faller inn under EØS-avtalen. Registeret skal gjøre det enklere for offentlige myndigheter, som vurderer å gi offentlig støtte, å få informasjon om foretakene har fått annen offentlig støtte som må tas med i vurderingen.",
                  },
                  "id": "40000eb2-8bc8-4b63-a371-411c3270b7a0",
                  "title": Object {
                    "nb": "Registeret for offentlig støtte",
                  },
                  "uri": "http://data.brreg.no/datakatalog/dataset/974760673/15",
                },
                Object {
                  "description": Object {
                    "en": "The Register of Company Accounts receives yearly about 315 000 annual accounts from limited companies, saving banks, insurance companies and other accounting obligation companies in Norway. The yearly accounts contains accounting numbers in electronic formats. In addition attachments includes the annula report of the board, and notes, cash flow and auditors report where needed. All yearly accounts is availiable for everyone, and the most important datasource for everyone that wish to get understand the financial situation for Norwegian businesses.",
                    "nb": "Regnskapsregisteret mottar årlig ca 315 000 (pr år 2017) årsregnskap fra aksjeselskaper, sparebanker, forsikringsselskaper og andre regnskapspliktige foretak i Norge. Årsregnskapene inneholder regnskapstall i elektronisk form. I tillegg er det vedlagt styrets årsberetning, noter, kontantstrømanalyse og revisors beretning, hvor dette kreves.",
                  },
                  "id": "1a3eda0a-4710-4209-8e9f-eb9d709f1540",
                  "title": Object {
                    "en": "Register of Company Accounts",
                    "nb": "Regnskapsregisteret",
                  },
                  "uri": "http://data.brreg.no/datakatalog/dataset/974760673/4",
                },
              ],
              "definition": Object {
                "no": "Et nisifret nummer som tildeles av Enhetsregisteret for en organisasjon som skal operere som en offentlig aktør",
              },
              "identifier": "https://data-david.github.io/Begrep/begrep/Organisasjonsnummer",
              "prefLabel": Object {
                "no": "organisasjonsnummer",
              },
              "uri": "https://data-david.github.io/Begrep/begrep/Organisasjonsnummer",
            },
          ]
        }
      />
      <div
        className="row"
      >
        <div
          className="col-md-6"
          key="remove-0https://data-david.github.io/Begrep/begrep/Enhet"
        >
          <button
            className="fdk-concepts-compare-rm-term fdk-button-small fdk-fullwidth mt-5 fdk-modal-button"
            onClick={[Function]}
          >
            <i
              className="fa fa-times fdk-color0"
            />
             
            Remove concept
          </button>
        </div>
        <div
          className="col-md-6"
          key="remove-1https://data-david.github.io/Begrep/begrep/Organisasjonsnummer"
        >
          <button
            className="fdk-concepts-compare-rm-term fdk-button-small fdk-fullwidth mt-5 fdk-modal-button"
            onClick={[Function]}
          >
            <i
              className="fa fa-times fdk-color0"
            />
             
            Remove concept
          </button>
        </div>
      </div>
    </ModalBody>
  </Modal>
  <div>
    <button
      className="bottom-modal-label fdk-button fdk-button-cta fdk-modal-button"
      onClick={[Function]}
    >
      <i
        className="fa fa-chevron-up fdk-color0"
      />
       
      You have added 2 concepts to comparison - click to show
    </button>
  </div>
</div>
`;
